File: solutions/leetcode_3075.py
class Solution:
    def maximumHappinessSum(self, happiness: list[int], k: int) -> int:
        happiness.sort(reverse=True)
        fee = 0
        result = 0
        for h in happiness:
            h -= fee
            if h <= 0:
                break
            result += h
            fee += 1
            if fee == k:
                break
        return result

File: solutions/leetcode_3075.java
import java.util.Arrays;

class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        int fee = 0;
        long result = 0;
        for (int i = happiness.length - 1; i >= 0; i--) {
            int h = happiness[i];
            h -= fee;
            if (h <= 0) {
                break;
            }
            result += h;
            fee += 1;
            if (fee == k) {
                break;
            }
        }
        return result;
    }
}
